src/pages/lessons/[slug].tsx

```diff
@@ -411,83 +433,16 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
   }, [currentPlayerState, session_id])
 
   React.useEffect(() => {
-    async function run() {
-      console.debug('loading video with auth')
-      const loadedLesson = await loadLesson(initialLesson.slug)
-      console.debug('authed video loaded', {video: loadedLesson})
-
-      send({
-        type: 'LOADED',
-        lesson: {...initialLesson, ...loadedLesson},
-        viewer,
-      })
-    }
-
-    if (cookieUtil.get(`egghead-watch-count`)) {
-      setWatchCount(Number(cookieUtil.get(`egghead-watch-count`)))
-    } else {
-      setWatchCount(
-        Number(
-          cookieUtil.set(`egghead-watch-count`, 0, {
-            expires: 15,
-          }),
-        ),
-      )
-    }
-
+    // when this effect is triggered, put the machine in the loading state (via
+    // `LOAD`). That state should take care of the rest with an Invoked
+    // Callback.
     send({
       type: 'LOAD',
       lesson: initialLesson,
       viewer,
     })
-
-    run()
   }, [initialLesson.slug])
 
-  // TODO: Temporary duplication of the above useEffect to ensure the lesson is
-  // loaded when transitioning from the 'subscribing'/Thank You overlay to
-  // watching the lesson. If this loading process was moved into the machine as
-  // an Invoked Callback, then state transitions could be employed to handle
-  // what the useEffects are doing.
-  React.useEffect(() => {
```

Comment:
This entire useEffect just happens in the machine as part of a transition into the `loading` state.